scripts/linters/pylint_extensions.py

```diff
@@ -1402,7 +1562,107 @@ def process_module(self, node):
                 elif not any(word in line for word in EXCLUDED_PHRASES):
                     self.add_message(
                         'no-newline-used-at-end', line=line_num + 1)
-                is_docstring = False
+                in_docstring = False
+                in_docstring_returns_section = False
```

Comment:
This isn't quite the way to model it, I think. Have a variable called current_docstring_section and allow it to take one of four values: None, DOCSTRING_SECTION_RETURNS, DOCSTRING_SECTION_RAISES, DOCSTRING_SECTION_YIELDS.

Reply:
Done